### PR TITLE
[doc only] Rewire the timespan docs to the new format

### PR DESCRIPTION
**Pull request by chutten**

### CODE REVIEW

**Review by brizental — APPROVED** [review]
r+wc! Thanks for taking this one :D

File: docs/user/reference/metrics/timespan.md [hunk 1]
```diff
@@ -45,383 +25,637 @@ fun onShowLogin() {
...
 
-def on_login_cancel():
-    metrics.auth.login_time.cancel()
-    # ...
+Auth.loginTime.measure {
+    // Process login flow
+}
 ```
+</div>
+<div data-lang="Java" class="tab"></div>
```
[the remark this answers is not in the document]
Comment (chutten) [reply docs/user/reference/metrics/timespan.md:299]: I have no idea! : D

(The source material didn't document it, so I assumed not. But maybe @Dexterp37 knows)

[the remark this answers is not in the document]
Comment (Dexterp37) [reply docs/user/reference/metrics/timespan.md:299]: Mh, I just tried and it seems to work:

```
        GleanBaseline.INSTANCE.duration().measure(() -> {
            System.out.println("Test");
            return null;
        });
```

with the exception that the `return` statement is required :O

[the remark this answers is not in the document]
Comment (chutten) [reply docs/user/reference/metrics/timespan.md:299]: Wait, these Java examples should have a `()` after `loginTime`? Like `Auth.INSTANCE.loginTime().cancel();`?

[the remark this answers is not in the document]
Comment (brizental) [reply docs/user/reference/metrics/timespan.md]: > Wait, these Java examples should have a () after loginTime? Like Auth.INSTANCE.loginTime().cancel();?

This is wrong everywhere else, heh.

[the remark this answers is not in the document]
Comment (Dexterp37) [reply docs/user/reference/metrics/timespan.md]: Yeah, [it is wrong](https://github.com/MozillaReality/FirefoxReality/blob/0432da7fbaa5a0cbc6a157f49bf94f49871f3e16/app/src/common/shared/org/mozilla/vrbrowser/telemetry/GleanMetricsService.java#L342) everywhere else I'm afraid :(

[the remark this answers is not in the document]
Comment (brizental) [reply docs/user/reference/metrics/timespan.md]: https://bugzilla.mozilla.org/show_bug.cgi?id=1712061 filed a bug...

**Pull request merged**